Assets/Scripts/UI/EMCli.cs

```diff
@@ -0,0 +1,184 @@
+﻿using UnityEngine;
+using System.Collections;
+using System;
+using Assets.Scripts.UI;
+using UnityEngine.SceneManagement;
+
+public class EMCli : MonoBehaviour
+{
+    private static Rect windowRect = new Rect(Screen.width - 320f, 0, 300f, 430f);
+    private static GUILayoutOption[] options1 = new GUILayoutOption[] { GUILayout.Width(320f), GUILayout.MaxHeight(430f) };
+    private static GUI.WindowFunction func1 = new GUI.WindowFunction(WindowLayout);
+    private static string layout = string.Empty;
+    private static Vector2 scrollPosition = Vector2.zero;
+    private static string inputLine = string.Empty;
+    public bool Visible = true;
+
+    // Use this for initialization
+    void Start()
+    {
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.F3))
+        {
+            Visible = !Visible;
+        }
+    }
+
+    void OnGUI()
+    {
+        if (Visible)
+        {
+            if (Event.current.type == EventType.KeyUp && Event.current.keyCode == KeyCode.RightAlt)
+            {
+                if (GUI.GetNameOfFocusedControl() == "CLI")
+                {
+                    GUI.FocusControl(string.Empty);
+                }
+                else
+                {
+                    GUI.FocusControl("CLI");
+                    //GUI.GetNameOfFocusedControl().SendCli();
+                }
+            }
+
+            EMCliGUI();
+        }
+    }
+
+    public static void AddLine(string line)
+    {
+        layout = string.Concat(layout, '\n', line);
+        scrollPosition += new Vector2(0, 150);
+    }
+
+    public void EMCliGUI()
+    {
+        GUI.backgroundColor = Color.green;
+        windowRect = GUILayout.Window(208, windowRect, func1, "EM Command Line", options1);
+    }
+    
+    public static void WindowLayout(int windowID)
+    {
+        GUI.backgroundColor = Color.green;
+        GUILayout.BeginHorizontal();
+        if (GUILayout.Button(">", GUILayout.MaxWidth(25)))
+        {
+            try
+            {
+                EnterCommand(inputLine);
+            }
+            catch (Exception ex)
```

Comment:
Rather catch a specific exception here. Commands will execute game logic. If the game logic has an exception, it should also appear within the unity error log.